blog/_posts/2023-04-18-integrating-scalajs-with-vite.md

```diff
@@ -0,0 +1,127 @@
+
+Building on ES Modules in this way provides a particularly nice opportunity for integration with Scala.js.
+Indeed, Scala.js can be configured to emit small ES Modules out of the Scala code.
+Even better, we can configure it to provide few large modules for dependencies (which rarely change), and small modules for the application code (which changes often).
+Assuming the application lives in the package `my.app`, the required configuration is
+
+```scala
+scalaJSLinkerConfig ~= {
+  _.withModuleKind(ModuleKind.ESModule)
+    .withModuleSplitStyle(
+      ModuleSplitStyle.SmallModulesFor(List("my.app")))
+},
+```
+
+When we edit and save a `.scala` file, a chain of incremental build steps is triggered:
+
+1. The Scala incremental compiler recompiles only that `.scala` file and produces the corresponding `.sjsir` files (the `.class` files of Scala.js).
+2. The Scala.js incremental linker picks up the new `.sjsir` files, incrementally re-optimizes the application, and rewrites only the target `.js` modules that have changed.
+3. Vite picks up the changed `.js` files and asks the browser to reload them.
+
+In order to reinforce that effect, we brought significant performance improvements to the Scala.js linker in [version 1.13.1](https://www.scala-js.org/news/2023/04/10/announcing-scalajs-1.13.1/), in particular in the incremental case.
+Together with all the steps above being fully incremental, updating the UI of your application stays sub-second.
+
+## Introducing vite-plugin-scalajs
+
+To configure Vite to pick up the output of Scala.js, in `fastLinkJS` for development mode and `fullLinkJS` for production mode, we released [`vite-plugin-scalajs`](https://github.com/scala-js/vite-plugin-scalajs).
+It is a plugin for Vite that directly connects it with a Scala.js sbt project (support for other Scala build tools will come later).
+
+After installing with
+
+```shell
+$ npm install -D @scala-js/vite-plugin-scalajs@1.0.0
+```
+
+and enabling it in `vite.config.js` as follows:
+
+```javascript
+import { defineConfig } from "vite";
+import scalaJSPlugin from "@scala-js/vite-plugin-scalajs";
+
+export default defineConfig({
+  plugins: [scalaJSPlugin()],
+});
+```
+
+we can import the output of Scala.js from `.js` or `.ts` files as follows:
+
+```javascript
+import 'scalajs:main.js';
+```
+
+`npm run dev`, the development mode of Vite, will automatically use the output of `fastLinkJS` in the Scala.js project.
+Its production commands `npm run build` and `npm run preview` will use the output of `fullLinkJS`.
+
+With this first version, a separate shell must be opened with sbt running `~fastLinkJS` for development mode.
+As the plugin gets more usage, we will enhance it so that it can keep the `~fastLinkJS` itself in the background.
+
+## New tutorial series
+
+The release of `vite-plugin-scalajs` is part of a larger goal to simplify the developer experience in Scala.js.
```

Comment:
I think this sentence would make a great second lead for the whole blog post, coming immediately after the opening 1 or 2 sentences.